pkg/agent/packetcapture/packetcapture_controller.go

```diff
@@ -449,7 +449,14 @@ func (c *Controller) performCapture(
 	if err != nil {
 		return false, err
 	}
-	pcapngWriter, err := pcapgo.NewNgWriter(file, layers.LinkTypeEthernet)
+
+	// set SnapLength here to make tcpdump on Mac OSX works. By default, its value is
+	// 0 and means unlimited, but tcpdump on Mac OSX will complain:
+	// 'tcpdump: pcap_loop: invalid packet capture length <len>, bigger than snaplen of 524288'
+	ngInterface := pcapgo.DefaultNgInterface
+	ngInterface.SnapLength = 65536
```

Comment:
do you think the snapshot length could be a parameter to the `Capture` interface method, after which `snapLength` could be defined as a constant in this file, and we could use the constant here, as well as when invoking `Capture`?

Reply:
sure, i will take a look.